frontend/src/integrations/groupsio/components/groupsio-connect-drawer.vue

```diff
@@ -0,0 +1,379 @@
+  form.groups.splice(index, 1);
+};
+
+const $v = useVuelidate(rules, form);
+
+const validateAccount = async () => {
+  isVerifyingAccount.value = true;
+  accountVerificationFailed.value = false;
+  try {
+    const response = await IntegrationService.groupsioGetToken(form.email, form.password, form.twoFactorCode);
+    const { groupsioCookie } = response;
+    cookie.value = groupsioCookie;
+    isAPIConnectionValid.value = true;
+  } catch (e) {
+    isAPIConnectionValid.value = false;
+    accountVerificationFailed.value = true;
+  }
+  isVerifyingAccount.value = false;
+};
+
+const reverifyAccount = () => {
+  isAPIConnectionValid.value = false;
+};
+
+const validateGroup = async (groupName) => {
+  try {
+    await IntegrationService.groupsioVerifyGroup(groupName, cookie.value);
+    return true;
+  } catch (e) {
+    return false;
+  }
+};
+
+const onBlurEmail = () => {
+  $v.value.email.$touch();
+  canVerify();
+};
+
+const onBlurPassword = () => {
+  $v.value.password.$touch();
+  canVerify();
+};
+
+const onBlurTwoFactorCode = () => {
+  $v.value.twoFactorCode.$touch();
+  canVerify();
+};
+
+const canVerify = async () => {
+  const isCorrect = await $v.value.$validate();
+  if (isCorrect) {
+    isVerificationEnabled.value = true;
+  } else {
+    isVerificationEnabled.value = false;
+  }
+};
+
+const emit = defineEmits(['update:modelValue']);
+
+const { hasFormChanged, formSnapshot } = formChangeDetector(form);
+
+const logoUrl = CrowdIntegrations.getConfig('groupsio').image;
+
+const isVisible = computed({
+  get() {
+    return props.modelValue;
+  },
+  set(value) {
+    return emit('update:modelValue', value);
+  },
+});
+
+const handleCancel = () => {
+  formSnapshot();
+  emit('update:modelValue', false);
+  if (!props.integration?.settings?.email) {
+    form.email = '';
+    form.password = '';
+    form.twoFactorCode = '';
+    form.groups = [''];
+    form.groupsValidationState = new Array(form.groups.length).fill(true);
+    cookie.value = '';
+    isAPIConnectionValid.value = false;
+    isVerifyingAccount.value = false;
+    accountVerificationFailed.value = false;
+    $v.value.$reset();
+  } else {
+    form.email = props.integration?.settings?.email;
+    form.password = '';
+    form.twoFactorCode = '';
+    // eslint-disable-next-line no-unsafe-optional-chaining
+    form.groups = props?.integration?.settings?.groups ? [...props.integration?.settings?.groups] : [''];
+    form.groupsValidationState = new Array(form.groups.length).fill(true);
+    cookie.value = props.integration?.settings?.token;
+    isAPIConnectionValid.value = true;
+    isVerifyingAccount.value = false;
+    accountVerificationFailed.value = false;
+    $v.value.$reset();
+  }
+};
+
+onMounted(() => {
+  if (props.integration?.settings?.email) {
+    form.email = props?.integration?.settings?.email;
+    // eslint-disable-next-line no-unsafe-optional-chaining
+    form.groups = [...props.integration?.settings?.groups];
```

Comment:
Maybe instead of disabling the eslint warning, only add form.groups if props.integration?.settings?.groups exist, inside an if

Reply:
Added ternary operator